### PR TITLE
AP_Avoidance: stop copying adsb vehicle onto stack in src_id_for_adsb…

### PR DESCRIPTION
**Pull request by peterbarker**
…_vehicle

### CODE REVIEW

File: libraries/AP_Avoidance/AP_Avoidance.h [hunk 1]
```diff
@@ -156,7 +156,7 @@ class AP_Avoidance {
     void deinit();
 
     // get unique id for adsb
-    uint32_t src_id_for_adsb_vehicle(AP_ADSB::adsb_vehicle_t vehicle) const;
+    uint32_t src_id_for_adsb_vehicle(const struct AP_ADSB::adsb_vehicle_t &vehicle) const;
```
Comment (rmackay9) [inline libraries/AP_Avoidance/AP_Avoidance.h]: it's slightly funny to have the "struct" in the declaration but not in the implementation.  I'd remove it from the declaration.

**Comment by peterbarker** [comment]
On Tue, 9 Jul 2019, Randy Mackay wrote:

> it's slightly funny to have the "struct" in the declaration but not in the
> implementation. I'd remove it from the declaration.

The struct declaration is helping with #include-loop issues.

**Comment by peterbarker** [comment]
On Thu, 11 Jul 2019, Pierre Kancir wrote:

> I am like Randy, I don't like to put the "struct" only in the declaration.
> How does it help with the include loop ? Shouldn't it better to fix the loop instead of using this hack ?

Yes, it's better to fix the loops.  Which is what several of the PRs I 
currently in place attempt to do.

**Comment by OXINARF** [comment]
I'm not sure what include loop issues we are having, but I'm generally in favor of forward declarations. If people are thrown off by the different method signature, I think it can be added to the definition too.

However, we should remove the header include - I see no reason to have it.

**Comment by peterbarker** [comment]
I've removed the change to add `struct` in; I expect it will come up on another PR, however :-)

**Pull request merged**